tests/e2e_test.go

```diff
@@ -294,15 +315,205 @@ var _ = ginkgo.BeforeSuite(func() {
+			Expect(err).Should(BeNil())
+
+			// Construct the transaction to send the Warp message to the destination chain
+			log.Info("Constructing transaction for the destination chain")
+			signedMessage, err := avalancheWarp.ParseMessage(signedWarpMessageBytes)
+			Expect(err).Should(BeNil())
+
+			numSigners, err := signedMessage.Signature.NumSigners()
+			Expect(err).Should(BeNil())
+
+			gasLimit, err := teleporter.CalculateReceiveMessageGasLimit(numSigners, defaultTeleporterMessage.RequiredGasLimit)
+			Expect(err).Should(BeNil())
+
+			callData, err := teleporter.EVMTeleporterContractABI.Pack("receiveCrossChainMessage", fundedAddress)
+			Expect(err).Should(BeNil())
+
+			baseFee, err := chainBRPCClient.EstimateBaseFee(ctx)
+			Expect(err).Should(BeNil())
+
+			gasTipCap, err := chainBRPCClient.SuggestGasTipCap(ctx)
+			Expect(err).Should(BeNil())
+
+			nonce, err := chainBRPCClient.NonceAt(ctx, fundedAddress, nil)
+			Expect(err).Should(BeNil())
+
+			gasFeeCap := baseFee.Mul(baseFee, big.NewInt(2))
+			gasFeeCap.Add(gasFeeCap, big.NewInt(2500000000))
+			destinationTx := predicateutils.NewPredicateTx(
+				chainBIDInt,
+				nonce,
+				&teleporterContractAddress,
+				gasLimit,
+				gasFeeCap,
+				gasTipCap,
+				big.NewInt(0),
+				callData,
+				types.AccessList{},
+				warp.ContractAddress,
+				signedMessage.Bytes(),
+			)
+
+			// Sign and send the transaction on the destination chain
+			signer := types.LatestSignerForChainID(chainBIDInt)
+			signedTxB, err := types.SignTx(destinationTx, signer, fundedKey)
+			Expect(err).Should(BeNil())
+
+			log.Info("Sending transaction to destination chain")
+			err = chainBRPCClient.SendTransaction(context.Background(), signedTxB)
+			Expect(err).Should(BeNil())
+
+			// Sleep to ensure the new block is published to the subscriber
+			time.Sleep(5 * time.Second)
+			receipt, err := chainBRPCClient.TransactionReceipt(ctx, signedTxB.Hash())
+			Expect(err).Should(BeNil())
+			Expect(receipt.Status).Should(Equal(types.ReceiptStatusSuccessful))
+		}
+	}()
+
+	time.Sleep(5 * time.Second) // Give the relayer goroutine a chance to subscribe
 	log.Info("Set up ginkgo before suite")
-})
 
-var _ = ginkgo.AfterSuite(func() {
-	log.Info("Running ginkgo after suite")
-	Expect(manager).ShouldNot(BeNil())
-	Expect(manager.TeardownNetwork()).Should(BeNil())
-	Expect(os.Remove(warpChainConfigPath)).Should(BeNil())
-})
+	setupStruct := SetupStruct{
+		Subnets: []SubnetStruct{
+			{
+				SubnetID:      subnetA,
+				ValidatorURIs: chainANodeURIs,
+				BlockchainID:  blockchainIDA,
+			},
+			{
+				SubnetID:      subnetB,
+				ValidatorURIs: chainBNodeURIs,
+				BlockchainID:  blockchainIDB,
+			},
+		},
+	}
+	setupBytes, err := json.Marshal(setupStruct)
+	Expect(err).Should(BeNil())
+
+	return setupBytes
+},
+	// Re-set all global variables that are referenced in the tests and set in the above function
+	func(setupBytes []byte) {
+		var setupStruct SetupStruct
```

Comment:
I think with the way you have the consolidate PR set up now, we can just do the necessary duplication in the getter function. We should just have to redial the client (or not, leave it up to the test to manage it's connections) and copy the *big.Int for the chainID